tests/data/data.ts

```diff
@@ -448,43 +449,7 @@ export class TestDataProvider {
     return retVal;
   }
 
-  static getAggregationPipeline(
-    matchConditionField: string,
-    matchConditionValue: string,
-    projection?: { [k: string]: number }
-  ) {
-    return [
-      // Stage 1: Unwind the repos array to create multiple documents for each referenced repo
-      {
-        $unwind: '$repos',
-      },
-      // Stage 2: Lookup to join with the repos_branches collection
-      {
-        $lookup: {
-          from: 'repos_branches',
-          localField: 'repos',
-          foreignField: '_id',
-          as: 'repo',
-        },
-      },
-      // Stage 3: Match documents based on given field
-      {
-        $match: {
-          [`repo.${matchConditionField}`]: matchConditionValue,
-        },
-      },
-      // Stage 4: Merge/flatten repo into docset
-      {
-        $replaceRoot: { newRoot: { $mergeObjects: [{ $arrayElemAt: ['$repo', 0] }, '$$ROOT'] } },
-      },
-      // Stage 5: Exclude fields
-      {
-        $project: projection || {
-          _id: 0,
-          repos: 0,
-          repo: 0,
-        },
-      },
-    ];
+  static getAggregationPipeline(matchConditions: { [k: string]: string }, projection?: { [k: string]: number }) {
```

Comment:
NIT: Could we just use `DocsetsRepository.getAggregationPipeline` in the tests, or is the wrapper needed?

Reply:
I was thinking this was more integrated, but you're right! Removed.